delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala

```diff
@@ -197,6 +197,60 @@ final class Files(
     } yield res
   }.span("createLink")
 
+  /**
+    * Create a file from a source file potentially in a different organization
+    * @param sourceId
+    *   File lookup id for the source file
+    * @param dest
+    *   Project, storage and file details for the file we're creating
+    */
+  def copyTo(
+      sourceId: FileId,
+      dest: CopyFileDestination
+  )(implicit c: Caller): IO[FileResource] = {
+    for {
+      (file, sourceDesc, sourceEntity) <- fetchSourceFile(sourceId)
+      (pc, storageRef, storage)        <- fetchDestinationStorage(dest)
+      _                                <- validateStorageTypeForCopy(file.storageType, storage)
+      iri                              <- dest.fileId.fold(generateId(pc))(FileId(_, dest.project).expandIri(fetchContext.onCreate).map(_._1))
+      destinationDesc                  <- FileDescription(dest.filename.getOrElse(sourceDesc.filename), sourceDesc.mediaType)
+      attributes                       <- saveFile(iri, storage, destinationDesc, sourceEntity)
+      res                              <- eval(CreateFile(iri, dest.project, storageRef, storage.tpe, attributes, c.subject, dest.tag))
+    } yield res
+  }.span("copyFile")
+
+  private def fetchSourceFile(id: FileId)(implicit c: Caller) =
+    for {
+      file                        <- fetch(id)
+      (iri, _)                    <- id.expandIri(fetchContext.onRead)
+      sourceStorage               <- storages.fetch(file.value.storage, id.project)
+      _                           <- validateAuth(id.project, sourceStorage.value.storageValue.readPermission)
+      attributes                   = file.value.attributes
+      sourceBytes                 <- fetchFile(sourceStorage.value, attributes, file.id)
```

Comment:
We should have to fetch the file itself to push it back.
For example for remote storage, it will mean get one file from one datacenter to another to push again to the first one.
Even for the local storage, we can just copy the file without having fetch the file to Delta.
Like we have a `SaveFile` and a `LinkFile`, we need a copy file to handle that.
That means also changes on the storage app to add a route for this copy operation

Reply:
Just to clarify: we _shouldn't_ have to fetch the file 😅